osbrain/agent.py

```diff
@@ -1668,10 +1672,13 @@ class AgentProcess(multiprocessing.Process):
     Agent class. Instances of an Agent are system processes which
     can be run independently.
     """
-    def __init__(self, name, nsaddr=None, addr=None, serializer=None,
+    def __init__(self, name=None, nsaddr=None, addr=None, serializer=None,
                  transport=None, base=Agent, attributes=None):
         super().__init__()
-        self.name = name
+        if name is None:
+            self.name = ''
+        else:
+            self.name = name
```

Comment:
You can revert this back to `self.name = name`. No need for this `if-else` I think.

Reply:
I needed to do it this way. Setting `self.name` to None raises an `AssertionError` from the multiprocessing library. Apparently, `name` must be a string.

Reply:
Yeah, forgot `name` is an actual parameter of the `multiptrocessing.Process` class.

Maybe you can just change the defaults to `''` instead of `None`. Would that work?

Reply:
Yeah, that's quite simpler, actually. And it seems to work just fine.